opentelemetry-otlp/src/retry.rs

```diff
@@ -144,13 +128,13 @@ where
 
                 match error_type {
                     RetryErrorType::NonRetryable => {
-                        otel_warn!(name: "OtlpRetry", message = format!("Operation {:?} failed with non-retryable error: {:?}", operation_name, err));
+                        otel_warn!(name: "OtlpRetryNonRetryable", operation = operation_name, error = format!("{:?}", err));
```

Comment:
otel_warn!(
                            name: "Export.Failed.NonRetryable",
                            signal: "Logs",
                            transport: "HTTP",
                            message = "OTLP export failed with non-retryable error - telemetry data will be lost"
                        );

^ Suggestion. We should probably include minimum info like erro_code too, but detailed error message should be done at an inner level, not by the retry module.

Reply:
I've taken the `name` style from here, and left the signal/transport out, favouring `operation`, based on my comment above (it tells you this stuff already). 

I am leery of adding `error_code` here, as it's generally logged anyway and it would make the interfaces around retry more complicated just for extra logging, but it sounds like you agree with that take ("detailed error message should be done at an inner level") ?

Reply:
got it. Yes `operation` sounds good enough.